app/web_framework/cluster_base_handler.py

```diff
@@ -62,6 +69,24 @@ def _handle_request_exception(self, ex):
         self.set_status(status_code)
         self.finish()
 
+    # For methods that annotate with `Tuple`, use `disable=invalid-sequence-index`
+    # This bug is fixed in https://github.com/PyCQA/astroid/commit/563031aaf13a44adc8db4f8d0ab8020d550aae00
+    # More information on the issue in https://github.com/PyCQA/pylint/issues/1212
+
+    def get_pagination_params(self) -> Tuple[int, int]:  # pylint: disable=invalid-sequence-index
+        offset = self.get_query_argument('offset', pagination_constants['DEFAULT_OFFSET'], True)
+        limit = self.get_query_argument('limit', pagination_constants['DEFAULT_LIMIT'], True)
+        return self._validate_arguments(int(offset), int(limit))
+
+    @staticmethod
+    def _validate_arguments(offset: int, limit: int) -> Tuple[int, int]:  # pylint: disable=invalid-sequence-index
+        offset = offset or pagination_constants['DEFAULT_OFFSET']
+        limit = limit or pagination_constants['DEFAULT_LIMIT']
+        # Negative offsets or limits are not allowed
+        non_negative_offset = max(offset, 0)
```

Comment:
I made this comment on an earlier commit, making it again to make sure you see it since github folded the old comment away with the older commit.

1. offset and limit are user provided strings, which means `int(some_string)` might raise a value error and return a 500 to the user. Catch the error and return a 400 instead.

2. Might as well just rename non_negative_offset and non_negative_limit to offset and limit.

3. If you go with my earlier suggestion (shamelessly stolen from Joey) about allowing offset and limit to be negative, you'll want to get rid of max logic from this method (allow negative numbers to be passed through to the `builds` method.)